packages/nodes-base/nodes/ActionNetwork/ActionNetwork.node.ts

```diff
@@ -0,0 +1,525 @@
+import {
+	IExecuteFunctions,
+} from 'n8n-core';
+
+import {
+	IDataObject,
+	INodeExecutionData,
+	INodeType,
+	INodeTypeDescription,
+} from 'n8n-workflow';
+
+import { createPersonSignupHelperFields, createFilterFields } from './UserInterface';
+import { actionNetworkApiRequestREST, constructODIFilterString, constructPersonSignupHelperPayload } from './GenericFunctions';
+
+export class ActionNetwork implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'Action Network',
+		name: 'actionNetwork',
+		icon: 'file:actionnetwork.svg',
+		group: ['output'],
+		version: 1,
+		subtitle: '={{$parameter["method"] + ": " + $parameter["resource"]}}',
+		description: 'Interact with an Action Network group\'s data',
+		defaults: {
+			name: 'Action Network',
+			color: '#9dd3ed',
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'ActionNetworkGroupApiToken',
+				required: true,
+			},
+		],
+		properties: [
+			{
+				displayName: 'API Version',
+				name: 'version',
+				type: 'options',
+				options: [{ name: 'Version 2', value: 'v2', },],
+				default: 'v2',
+				description: 'API version',
+			},
+			{
+				displayName: 'Resource',
+				name: 'resource',
+				type: 'options',
+				options: [
+					{ name: 'Person', value: 'people', },
+					{ name: 'Petition', value: 'petitions' },
+					{ name: 'Event', value: 'events' },
+					{ name: 'Form', value: 'forms' },
+					{ name: 'Fundraising Page', value: 'fundraising_pages' },
+					{ name: 'Event Campaign', value: 'event_campaigns' },
+					{ name: 'Campaign', value: 'campaigns' },
+					{ name: "Message", value: 'messages' }
+				],
+				default: 'people',
+				description: 'The resource to work with.',
+			},
+			{
+				displayName: 'Resource ID',
+				name: 'resource_id',
+				type: 'string',
+				required: false,
+				default: '',
+				displayOptions: {
+					show: {
+						resource: ['events', 'petitions', 'forms', 'fundraising_pages', 'campaigns'],
+						method: ['POST', 'GET']
+					}
+				}
+			},
+			//
+			{
+				displayName: 'Related Resource',
+				name: 'related_resource',
+				type: 'options',
+				options: [
+					{ name: 'N/A', value: '' },
+					{ name: 'Attendance', value: 'attendances' },
+				],
+				required: false,
+				default: '',
+				description: 'A resource related to its parent resource',
+				displayOptions: { show: { resource: ['events'] }, hide: { resource_id: [''] } }
+			},
```

Comment:
I'm wondering whether to invert this to make it more simple: doing away with `related_resource` and having one big list of `resource`s that then conditionally request parent resource IDs when needed. The code would become longer and more confusing though, possibly lots of if statements, so I'm unsure what best practice is here. Recommendations welcome.